File: src/com/colacelli/ircbot/plugins/autojoin/AutoJoinPlugin.java
package com.colacelli.ircbot.plugins.autojoin;

import com.colacelli.ircbot.IRCBot;
import com.colacelli.ircbot.Plugin;
import com.colacelli.irclib.actors.Channel;
import com.colacelli.irclib.connection.listeners.OnConnectListener;

import java.util.ArrayList;

public class AutoJoinPlugin implements Plugin {
    private ArrayList<Channel> channels;
    private OnConnectListener listener;

    public AutoJoinPlugin(ArrayList<Channel> channels) {
        this.channels = channels;

        listener = (connection, server, user) -> channels.forEach((channel) -> connection.join(channel));
    }

    @Override
    public String getName() {
        return "AUTO_JOIN";
    }

    @Override
    public void onLoad(IRCBot bot) {
        bot.addListener(listener);
    }

    @Override
    public void onUnload(IRCBot bot) {
        bot.removeListener(listener);
    }
}

File: src/com/colacelli/ircbot/plugins/autojoin/AutoJoinPlugin.kt
package com.colacelli.ircbot.plugins.autojoin

import com.colacelli.ircbot.IRCBot
import com.colacelli.ircbot.base.Plugin
import com.colacelli.irclib.actors.Channel
import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.connection.Server
import com.colacelli.irclib.connection.listeners.OnConnectListener

class AutoJoinPlugin(channels : ArrayList<Channel>) : Plugin {
    private var listener = object : OnConnectListener {
        override fun onConnect(connection: Connection, server: Server, user: User) {
            return channels.forEach {
                connection.join(it)
            }
        }
    }

    override var name = "auto_join"

    override fun onLoad(bot: IRCBot) {
        bot.addListener(listener)
    }

    override fun onUnload(bot: IRCBot) {
        bot.removeListener(listener)
    }
}